site/app/controllers/PollController.php

```diff
@@ -163,10 +163,25 @@ public function addNewPoll() {
         $poll_id = $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
         $file_path = null;
         if (isset($_FILES['image_file']) && $_FILES["image_file"]["name"] !== "") {
-            $file = $_FILES["image_file"];
-            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
-            move_uploaded_file($file["tmp_name"], $file_path);
-            $this->core->getQueries()->setPollImage($poll_id, $file_path);
+            $status = FileUtils::validateUploadedFiles($_FILES["image_file"]);
+            // validate the uploaded file size
+            if (array_key_exists("failed", $status)) {
```

Comment:
The array key `failed` will only be set if 0 files are sent to be validated which is taken care of the `isset` above. You can check if anything went wrong with `if( ! $status['success'])` and then use `$status['error']` to get an error message.